packages/tx/src/EIP2930Transaction.ts

```diff
@@ -0,0 +1,385 @@
+import {
+  Address,
+  BN,
+  bnToHex,
+  bnToRlp,
+  bufferToHex,
+  ecrecover,
+  keccak256,
+  rlp,
+  toBuffer,
+} from 'ethereumjs-util'
+import { BaseTransaction } from './baseTransaction'
+import {
+  AccessList,
+  AccessListBuffer,
+  AccessListItem,
+  isAccessList,
+  JsonTx,
+  TxData,
+  TxOptions,
+} from './types'
+
+// secp256k1n/2
+const N_DIV_2 = new BN('7fffffffffffffffffffffffffffffff5d576e7357a4501ddfe92f46681b20a0', 16)
+
+type EIP2930ValuesArray = [
+  Buffer,
+  Buffer,
+  Buffer,
+  Buffer,
+  Buffer,
+  Buffer,
+  Buffer,
+  AccessListBuffer,
+  Buffer?,
+  Buffer?,
+  Buffer?
+]
+
+export class EIP2930Transaction extends BaseTransaction<EIP2930Transaction> {
+  public readonly chainId: BN
+  public readonly accessList: AccessListBuffer
+  public readonly v?: BN
+  public readonly r?: BN
+  public readonly s?: BN
+
+  get transactionType(): number {
+    return 1
+  }
+
+  public readonly AccessListJSON: AccessList
+
+  // EIP-2930 alias for `s`
+  get senderS() {
+    return this.s
+  }
+
+  // EIP-2930 alias for `r`
+  get senderR() {
+    return this.r
+  }
+
+  // EIP-2930 alias for `v`
+
+  get yParity() {
+    return this.v
+  }
+
+  public static fromTxData(txData: TxData, opts: TxOptions = {}) {
+    return new EIP2930Transaction(txData, opts)
+  }
+
+  // Instantiate a transaction from the raw RLP serialized tx. This means that the RLP should start with 0x01.
+  public static fromRlpSerializedTx(serialized: Buffer, opts: TxOptions = {}) {
```

Comment:
This is actually a bit misleading. This method states that it should be an RlpSerialized transaction - but you actually have to input an invalid RLP (buffer starts with 0x01). I am not super sure here what to do, I like to have these static constructors with the same name on both this class and on LegacyTransaction. Thoughts?

Reply:
hmmm that's interesting, I do see the value in keeping the same name as Legacy, however if you are technically correct maybe it could be named `fromSerializedTx` and add `fromRlpSerializedTx` as an alias for compatibility.

Reply:
Just added this on both transactions!